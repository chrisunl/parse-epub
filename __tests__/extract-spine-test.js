import { contentOpf } from './data.js'
import extractSpine from '../extract/spine.js'

test('#extractSpine', () => {
  expect(extractSpine(contentOpf)).toMatchSnapshot();
});
