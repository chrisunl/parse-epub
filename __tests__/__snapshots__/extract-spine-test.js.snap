exports[`test #extractSpine 1`] = `
Object {
  "byId": Object {
    "item-1377": Object {
      "id": "item-1377",
      "linear": false,
      "properties": undefined,
    },
    "item-1471": Object {
      "id": "item-1471",
      "linear": false,
      "properties": undefined,
    },
    "item-data-uuid-01eee42463c54f1b945502ebf51ab9ab": Object {
      "id": "item-data-uuid-01eee42463c54f1b945502ebf51ab9ab",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-0cf7f56918514bb784e1de2b5e739da1": Object {
      "id": "item-data-uuid-0cf7f56918514bb784e1de2b5e739da1",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-0f371090c0e84cd09a06f868c3fe8efd": Object {
      "id": "item-data-uuid-0f371090c0e84cd09a06f868c3fe8efd",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-0f7b7bf0ed1b4c5c88b86bf81c8e2f92": Object {
      "id": "item-data-uuid-0f7b7bf0ed1b4c5c88b86bf81c8e2f92",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-101710cd602a4fcfa90108c5c0347df7": Object {
      "id": "item-data-uuid-101710cd602a4fcfa90108c5c0347df7",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-107c288bb253478f8a3094fdd57e00d0": Object {
      "id": "item-data-uuid-107c288bb253478f8a3094fdd57e00d0",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-14c6d82a58e64980992078f1726dac96": Object {
      "id": "item-data-uuid-14c6d82a58e64980992078f1726dac96",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-1695c0869dbf4fc38e255f1d8571b4e5": Object {
      "id": "item-data-uuid-1695c0869dbf4fc38e255f1d8571b4e5",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-17a19334765e441d848dd98a07d3b583": Object {
      "id": "item-data-uuid-17a19334765e441d848dd98a07d3b583",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-3a07ddda3070483e99e2b20349853c76": Object {
      "id": "item-data-uuid-3a07ddda3070483e99e2b20349853c76",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-3a5daf183fc34c31822a2b16d13a57ba": Object {
      "id": "item-data-uuid-3a5daf183fc34c31822a2b16d13a57ba",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-3b13d68a9f9447fe8bd6f2d7c7ea7403": Object {
      "id": "item-data-uuid-3b13d68a9f9447fe8bd6f2d7c7ea7403",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-3c0bb8315ea14a2bac12ace68240caff": Object {
      "id": "item-data-uuid-3c0bb8315ea14a2bac12ace68240caff",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-40c45e21a58c438283bf95dcaa809537": Object {
      "id": "item-data-uuid-40c45e21a58c438283bf95dcaa809537",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-445da3f24e8d49d897c41c17a615bcef": Object {
      "id": "item-data-uuid-445da3f24e8d49d897c41c17a615bcef",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-47a9d3e21464491cb37d718162b9b520": Object {
      "id": "item-data-uuid-47a9d3e21464491cb37d718162b9b520",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-553be8a225384235b06f21698e403a16": Object {
      "id": "item-data-uuid-553be8a225384235b06f21698e403a16",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-589d9984380e4568b1eb5710e6e81ca2": Object {
      "id": "item-data-uuid-589d9984380e4568b1eb5710e6e81ca2",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-5adbebda98604da48e98180467c64cfd": Object {
      "id": "item-data-uuid-5adbebda98604da48e98180467c64cfd",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-5b23866eff924613af22d8a296c61653": Object {
      "id": "item-data-uuid-5b23866eff924613af22d8a296c61653",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-601228182dbc40e2ba71787be4c05654": Object {
      "id": "item-data-uuid-601228182dbc40e2ba71787be4c05654",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-61380725c0dd484593df2807e84cf853": Object {
      "id": "item-data-uuid-61380725c0dd484593df2807e84cf853",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-6ad2f32297f84d1da04638f7e582c374": Object {
      "id": "item-data-uuid-6ad2f32297f84d1da04638f7e582c374",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-6d74e8e547ec455a9c00ebb37cb4a872": Object {
      "id": "item-data-uuid-6d74e8e547ec455a9c00ebb37cb4a872",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-74687195b0fb413bba6b9ca5d8aa3cec": Object {
      "id": "item-data-uuid-74687195b0fb413bba6b9ca5d8aa3cec",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-757f293b2be647deaa1c5291da00413e": Object {
      "id": "item-data-uuid-757f293b2be647deaa1c5291da00413e",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-7841571fa25b470fa9b4158ce0544f87": Object {
      "id": "item-data-uuid-7841571fa25b470fa9b4158ce0544f87",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-7bf20da5723c4a5799c86f97c2fd81b3": Object {
      "id": "item-data-uuid-7bf20da5723c4a5799c86f97c2fd81b3",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-8fc059832a1f4c64be19490536ebc955": Object {
      "id": "item-data-uuid-8fc059832a1f4c64be19490536ebc955",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-999b44bf3f03460cbdf41781bd702168": Object {
      "id": "item-data-uuid-999b44bf3f03460cbdf41781bd702168",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-9a306d7babff42c084fa63ad29079c97": Object {
      "id": "item-data-uuid-9a306d7babff42c084fa63ad29079c97",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-a4f73db0c4ff4464afdbdef6540a1feb": Object {
      "id": "item-data-uuid-a4f73db0c4ff4464afdbdef6540a1feb",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-ad4d37cb6ecb4c11ada0eafc3c44146f": Object {
      "id": "item-data-uuid-ad4d37cb6ecb4c11ada0eafc3c44146f",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-bda8bcd0fdaa4c26be8914772d6c84ec": Object {
      "id": "item-data-uuid-bda8bcd0fdaa4c26be8914772d6c84ec",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-cc1ee917649f4f55855784812d45ef9d": Object {
      "id": "item-data-uuid-cc1ee917649f4f55855784812d45ef9d",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-cc3d8d64c26046bc92151bfe70170c4c": Object {
      "id": "item-data-uuid-cc3d8d64c26046bc92151bfe70170c4c",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-d11317bd1e344490a568604ca1f4616f": Object {
      "id": "item-data-uuid-d11317bd1e344490a568604ca1f4616f",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-d15b805965d14521aeb48d93d168b65f": Object {
      "id": "item-data-uuid-d15b805965d14521aeb48d93d168b65f",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-d17e58059b0542f1a831de1593127652": Object {
      "id": "item-data-uuid-d17e58059b0542f1a831de1593127652",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-d1dc38f40de742b5a24353c05ed1a23e": Object {
      "id": "item-data-uuid-d1dc38f40de742b5a24353c05ed1a23e",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-d2a6e8f4db704bae875c9b4c5dc209cc": Object {
      "id": "item-data-uuid-d2a6e8f4db704bae875c9b4c5dc209cc",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-d43a2540b0774b5f973c841873a8b814": Object {
      "id": "item-data-uuid-d43a2540b0774b5f973c841873a8b814",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-debfcf17707946f0bf220b7924adaa48": Object {
      "id": "item-data-uuid-debfcf17707946f0bf220b7924adaa48",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-e8573f88150047078fdb5bfb083ca5c7": Object {
      "id": "item-data-uuid-e8573f88150047078fdb5bfb083ca5c7",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-ea0242f6d23d4e608c7252ad571882f5": Object {
      "id": "item-data-uuid-ea0242f6d23d4e608c7252ad571882f5",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-ec172a4c778d409da1aec2f5474cc337": Object {
      "id": "item-data-uuid-ec172a4c778d409da1aec2f5474cc337",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-f7923d60c5894cc999346a9f2c8f5ae7": Object {
      "id": "item-data-uuid-f7923d60c5894cc999346a9f2c8f5ae7",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-fb124defa1ef4c04a4f79397ae047df0": Object {
      "id": "item-data-uuid-fb124defa1ef4c04a4f79397ae047df0",
      "linear": true,
      "properties": undefined,
    },
    "item-data-uuid-ffc2e9c0ab5a404980a1c2686659ab6f": Object {
      "id": "item-data-uuid-ffc2e9c0ab5a404980a1c2686659ab6f",
      "linear": true,
      "properties": undefined,
    },
  },
  "items": Array [
    "item-1471",
    "item-data-uuid-3b13d68a9f9447fe8bd6f2d7c7ea7403",
    "item-data-uuid-d2a6e8f4db704bae875c9b4c5dc209cc",
    "item-data-uuid-6d74e8e547ec455a9c00ebb37cb4a872",
    "item-data-uuid-5b23866eff924613af22d8a296c61653",
    "item-data-uuid-40c45e21a58c438283bf95dcaa809537",
    "item-data-uuid-999b44bf3f03460cbdf41781bd702168",
    "item-data-uuid-01eee42463c54f1b945502ebf51ab9ab",
    "item-data-uuid-d17e58059b0542f1a831de1593127652",
    "item-data-uuid-6ad2f32297f84d1da04638f7e582c374",
    "item-data-uuid-5adbebda98604da48e98180467c64cfd",
    "item-data-uuid-9a306d7babff42c084fa63ad29079c97",
    "item-data-uuid-7bf20da5723c4a5799c86f97c2fd81b3",
    "item-data-uuid-47a9d3e21464491cb37d718162b9b520",
    "item-data-uuid-107c288bb253478f8a3094fdd57e00d0",
    "item-data-uuid-d15b805965d14521aeb48d93d168b65f",
    "item-data-uuid-0cf7f56918514bb784e1de2b5e739da1",
    "item-data-uuid-3c0bb8315ea14a2bac12ace68240caff",
    "item-data-uuid-a4f73db0c4ff4464afdbdef6540a1feb",
    "item-data-uuid-ffc2e9c0ab5a404980a1c2686659ab6f",
    "item-data-uuid-fb124defa1ef4c04a4f79397ae047df0",
    "item-data-uuid-d43a2540b0774b5f973c841873a8b814",
    "item-data-uuid-14c6d82a58e64980992078f1726dac96",
    "item-data-uuid-debfcf17707946f0bf220b7924adaa48",
    "item-data-uuid-d1dc38f40de742b5a24353c05ed1a23e",
    "item-data-uuid-ea0242f6d23d4e608c7252ad571882f5",
    "item-data-uuid-17a19334765e441d848dd98a07d3b583",
    "item-data-uuid-0f371090c0e84cd09a06f868c3fe8efd",
    "item-data-uuid-553be8a225384235b06f21698e403a16",
    "item-data-uuid-cc3d8d64c26046bc92151bfe70170c4c",
    "item-data-uuid-e8573f88150047078fdb5bfb083ca5c7",
    "item-data-uuid-601228182dbc40e2ba71787be4c05654",
    "item-data-uuid-d11317bd1e344490a568604ca1f4616f",
    "item-data-uuid-0f7b7bf0ed1b4c5c88b86bf81c8e2f92",
    "item-data-uuid-ad4d37cb6ecb4c11ada0eafc3c44146f",
    "item-data-uuid-3a5daf183fc34c31822a2b16d13a57ba",
    "item-data-uuid-61380725c0dd484593df2807e84cf853",
    "item-data-uuid-445da3f24e8d49d897c41c17a615bcef",
    "item-data-uuid-74687195b0fb413bba6b9ca5d8aa3cec",
    "item-data-uuid-757f293b2be647deaa1c5291da00413e",
    "item-data-uuid-f7923d60c5894cc999346a9f2c8f5ae7",
    "item-data-uuid-101710cd602a4fcfa90108c5c0347df7",
    "item-data-uuid-8fc059832a1f4c64be19490536ebc955",
    "item-data-uuid-cc1ee917649f4f55855784812d45ef9d",
    "item-data-uuid-3a07ddda3070483e99e2b20349853c76",
    "item-data-uuid-bda8bcd0fdaa4c26be8914772d6c84ec",
    "item-data-uuid-589d9984380e4568b1eb5710e6e81ca2",
    "item-data-uuid-1695c0869dbf4fc38e255f1d8571b4e5",
    "item-data-uuid-7841571fa25b470fa9b4158ce0544f87",
    "item-data-uuid-ec172a4c778d409da1aec2f5474cc337",
    "item-1377",
  ],
}
`;
